app/main.py

```
@@ -1,5 +1,56 @@
+class BaseRobot:
+    def __init__(self, name: str, weight: float, coords: list = None) -> None:
+        self.name = name
+        self.weight = weight
+        self.coords = coords if coords is not None else [0, 0]
+
+    def go_forward(self, step: int = 1) -> None:
+        self.coords[1] += step
+
+    def go_back(self, step: int = 1) -> None:
+        self.coords[1] -= step
+
+    def go_right(self, step: int = 1) -> None:
+        self.coords[0] += step
+
+    def go_left(self, step: int = 1) -> None:
+        self.coords[0] -= step
+
+    def get_info(self) -> str:
+        return f"Robot: {self.name}, Weight: {self.weight}"
+
+
+class FlyingRobot(BaseRobot):
+    def __init__(self, name: str, weight: float, coords: list = None) -> None:
+        if coords is None:
+            coords = [0, 0, 0]
+        else:
+            coords = (coords + [0, 0])[:3]
+        super().__init__(name, weight, coords)
+
+    def go_up(self, step: int = 1) -> None:
+        self.coords[2] += step
+
+    def go_down(self, step: int = 1) -> None:
+        self.coords[2] -= step
+
+
 class Cargo:
-    def __init__(self, weight: int) -> None:
+    def __init__(self, weight: float) -> None:
         self.weight = weight
 
-# write your code here
+
+class DeliveryDrone(FlyingRobot):
+    def __init__(self, name: str, weight: float,
+                 coords: list = None, max_load_weight: float = 0.0,
+                 current_load: Cargo = None) -> None:
+        super().__init__(name, weight, coords)
+        self.max_load_weight = max_load_weight
+        self.current_load = current_load
+
+    def hook_load(self, cargo: Cargo) -> None:
+        if self.current_load is None and cargo.weight <= self.max_load_weight:
+            self.current_load = cargo
```

Recommendation: It would be a good idea to add some feedback for the user in case the load can't be hooked. You can raise an exception or return a value to indicate that the load was too heavy or that there's already a load hooked.